factor_analyzer/factor_analyzer.py

```diff
@@ -114,6 +116,70 @@ def calculate_bartlett_sphericity(x):
     return statistic, p_value
 
 
+def calculate_cng_indices(
+    data: np.ndarray, model: str = "components"
+) -> Tuple[int, pd.DataFrame]:
+    """Calculate the Cattel-Nelson-Gorsuch indices, which are used to determine
+    the appropriate number of factors for a factor analysis.
+
+    Direct port of nCng function from nFactors package:
+    https://rdrr.io/cran/nFactors/man/nCng.html
+
+    Parameters
+    ----------
+    data : array-like
+        The array of samples x observable for which to calculate CNG indices
+    model : str
+        "components" or "factors"
+
+    Returns
+    -------
+    num_factors : int
+        The number of components/factors to retain
+    details : pd.DataFrame
+        The eigenvalues and CNG indices of the dataset
+    """
+    data = corr(data)
+    if model == "factors":
+        data -= np.linalg.pinv(np.diag(np.diag(np.linalg.pinv(data))))
+        # TODO: Should this line be here?
+        data = covariance_to_correlation(data)
+
+    values = np.sort(np.linalg.eigvals(data))[::-1]
+
+    num_variables = len(data)
```

Comment:
Won't `len(data)` give you the number of observations, rather than the number of variables, assuming `data` is a `numpy` array or `pandas` data frame?

Unfortunately, this doesn't translate exactly from R:
```{R}
> df <- data.frame(A = c(1, 2, 3), B = c(1, 2, 3))
> length(df) 
[1] 2
```

```{python}
>>> import pandas as pd
>>> df = pd.DataFrame({'A': [1, 2, 3], 'B': [1, 2, 3]})
>>> len(df)
3
>>> len(df.values)
3
```
Also, should we move the line below to the beginning?

Reply:
At this line of the code `data` refers to the correlation matrix, which is square. `data.shape[0] == data.shape[1]`